docs/operations/hook-sidecar.md

```diff
@@ -0,0 +1,145 @@
+# Hook Sidecar Container
+
+## Introduction
+
+In KubeVirt, a Hook Sidecar container is a sidecar container (a secondary container that runs along with the main
+application container within the same Pod) used to apply changes before the Virtual Machine is initialized. This
+ability is provided since 100% of the [libvirt domain XML](https://libvirt.org/formatdomain.html) elements and
+attributes are not configurable/customizable using the VMI specification.
+
+The sidecar containers communicate with the main container over a socket with a gRPC protocol. It is meant to do the
+changes over libvirt's XML and return the new XML over gRPC for the VM creation.
```

Comment:
> It is meant to do the changes over libvirt's XML and return the new XML over gRPC for the VM creation.

This refers only to the `onDefineDomain` hook but we also have the `preCloudInitIso` which does not requite libvirt's domain changes.

Reply:
Would it be right to describe `preCloudInitIso` hook as one that modifies the cloud-init data before spinning up the VMI?

Reply:
Right, but I think you can't escape mentioning/reference to [cloud-init](https://github.com/kubevirt/kubevirt/tree/main/pkg/cloud-init) (... which is missing a README.md file!)